### PR TITLE
blis, amdblis: remove custom phases, add v0.9.0 for blis

### PR DESCRIPTION
**Pull request by alalazo**
Also refactor a few variants

### CODE REVIEW

File: var/spack/repos/builtin/packages/blis/package.py [hunk 1]
```diff
@@ -9,38 +9,19 @@
 # https://github.com/flame/blis/issues/197
 
 
-class BlisBase(Package):
+class BlisBase(MakefilePackage):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blis/package.py:12]: FWIW. I was going to ask why you didn't switch to `AutotoolsPackage` so you could take advantage of some of the helper methods below until I read https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#important-files and looked at the `configure` script in the software repository.

File: var/spack/repos/builtin/packages/blis/package.py [hunk 2]
```diff
@@ -141,6 +101,7 @@ class Blis(BlisBase):
     git = "https://github.com/flame/blis.git"
 
     version('master', branch='master')
+    version('0.9.0', sha256='1135f664be7355427b91025075562805cdc6cc730d3173f83533b2c5dcc2f308')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/blis/package.py:104]: Confirmed

**Review by tldahlgren — APPROVED** [review]
LGTM.

**Comment by tldahlgren** [comment]
Deferring to package maintainers for the merge.

File: var/spack/repos/builtin/packages/blis/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/blis/package.py:12]: Yep, it seems a lot of people like to have custom scripts called `configure` :laughing:

**Comment by alalazo** [comment]
@amd-toolchain-support I'll go ahead with merging this, since it doesn't change much in `amdblis` (just the `lib` variant to be precise). Let me know if there are adjustments needed, I'll follow with another PR.

**Pull request merged**